src/Inputs/Slider.tsx

```diff
@@ -1,79 +1,440 @@
+
+    const MaxAndMinDifference = max - min;
+
+    let draggingThumb = 'Finish';
+
+    // Translate a value to Pixel
+    const translateToPixels = (theValue: number): number => {
+        const pixelTranslator =
+            (bar.current?.clientWidth as number) / MaxAndMinDifference;
+        return ((theValue as number) - min) * pixelTranslator;
+    };
+
+    // Translate pixels to a value and rounding up/down to steps
+    const translateToValue = (theValue: number): number => {
+        return (
+            Math.round(
+                ((theValue * MaxAndMinDifference) /
+                    (bar.current?.clientWidth as number) +
+                    min) /
+                    step,
+            ) * step
+        );
+    };
+
+    // thumb positions in values whenever it changes (by pixels)
+    const finishThumbLeftInValue = useMemo((): number => {
+        const Value = finishThumbLeft
+            ? translateToValue(finishThumbLeft)
+            : valueFinish;
+        return Value;
+    }, [finishThumbLeft]);
+
+    const startThumbLeftInValue = useMemo((): number => {
+        const Value = startThumbLeft
+            ? translateToValue(startThumbLeft)
+            : valueStart;
+        return Value;
+    }, [startThumbLeft]);
+
+    // Final Result
+    useMemo((): void => {
+        onChange({
+            values: {
+                lowValue: startThumbLeftInValue,
+                highValue: finishThumbLeftInValue,
+            },
+        });
+    }, [finishThumbLeftInValue, startThumbLeftInValue]);
+
+    // Calcs of Positioning a thumb in the specific scale with provided steps
+    const calculatePosition = (theValue: number): number => {
+        return translateToPixels(translateToValue(theValue));
+    };
+
+    useLayoutEffect((): void => {
+        if (bar.current) {
+            // setting up initial positions
+            if (valueFinish) {
+                setFinishThumbLeft(translateToPixels(valueFinish));
+            }
+            if (valueStart) {
+                setStartThumbLeft(translateToPixels(valueStart));
+            }
+
+            // placing the marks
+            if (marksBar.current) {
+                const theMarks = marksBar.current.children as any;
+                theMarks.forEach((child: HTMLElement): void => {
+                    child.style.left = `${translateToPixels(
+                        parseInt(child.style.left, 10),
+                    )}px`;
+                });
+            }
+        }
+    }, []);
+
+    // repositions marks and thumbs when resizing
+    function resizing(): void {
+        if (finishThumbLeftInValue) {
+            setFinishThumbLeft(translateToPixels(finishThumbLeftInValue));
+        }
+        if (startThumbLeftInValue) {
+            setStartThumbLeft(translateToPixels(startThumbLeftInValue));
+        }
+
+        if (marksBar.current && marks) {
+            const theMarks = marksBar.current.children as any;
```

Comment:
children are ReactNode ?

Reply:
When I put reactnode I get the ts error that reactnode is not assignable to htmlelement... If I change child type I get error that child.styles does not exist on string...